test/AccountSystem.t.sol

```diff
@@ -0,0 +1,1718 @@
+        uint256 sessionID,
+        uint256 expiration,
+        uint256 privateKey
+    ) internal view returns (bytes memory) {
+        DegenCasinoAccount account = DegenCasinoAccount(
+            payable(accountAddress)
+        );
+
+        bytes32 sessionHash = account.sessionHash(
+            executor,
+            sessionID,
+            expiration
+        );
+
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(privateKey, sessionHash);
+        return abi.encodePacked(r, s, v);
+    }
+
+    function test_play_with_valid_signatures() public {
+        // Create and fund the account
+        (address accountAddress, ) = accountSystem.createAccount(player2);
+        DegenCasinoAccount account = DegenCasinoAccount(
+            payable(accountAddress)
+        );
+
+        // Set up the game action
+        Action memory action = Action({
+            game: address(game),
+            data: abi.encodeWithSignature(
+                "mintGambit(address,uint256)",
+                accountAddress,
+                1 ether
+            ),
+            value: 0,
+            request: 1
+        });
+
+        // Set up executor terms
+        address[] memory rewardTokens = new address[](1);
+        rewardTokens[0] = address(game);
+        uint16[] memory basisPoints = new uint16[](1);
+        basisPoints[0] = 1000; // 10%
+        ExecutorTerms memory terms = ExecutorTerms({
+            rewardTokens: rewardTokens,
+            basisPoints: basisPoints
+        });
+
+        // Sign the session and terms
+        bytes memory sessionSig = _signSession(
+            accountAddress,
+            player1,
+            1,
+            block.timestamp + 1 hours,
+            player2PrivateKey
+        );
+        bytes memory termsSig = _signTerms(
+            accountAddress,
+            terms,
+            player2PrivateKey
+        );
+
+        // Execute playInSession
+        vm.prank(player1); // player1 is the executor
+        account.playInSession(
+            action,
+            terms,
+            1,
+            block.timestamp + 1 hours,
+            sessionSig,
+            termsSig
+        );
+
+        // Verify results
+        assertEq(game.balanceOf(accountAddress), 0.9 ether); // 90% of minted amount
+        assertEq(game.balanceOf(player1), 0.1 ether); // 10% reward to executor
+    }
+
+    function test_play_with_jackpot_win() public {
+        // Create and fund the account
+        (address accountAddress, ) = accountSystem.createAccount(player2);
+        DegenCasinoAccount account = DegenCasinoAccount(
+            payable(accountAddress)
+        );
+
+        // Fund the account for playing
+        vm.deal(accountAddress, 100 * game.CostToSpin());
+        vm.deal(address(game), 100 ether); // Ensure game has enough balance
+
+        // First action: spin
+        Action memory spinAction = Action({
+            game: address(game),
+            data: abi.encodeWithSignature("spin(bool)", false),
```

Comment:
Saw there wasn't any test for using gambit from the degen account system. This would be something we need to test.